Revise skip torchrec logic

Microve opened the pull request:
Summary:
The previous logic adds skipped files when the file was imported which happens at very early stage. However, we could set skip_torchrec at later stage (e.g, in APS, we set it during the trainer execution). In that case, the skip logic will still take effect since skipped files have been added.

So in this diff, we revise the logic so that it can adapt to changes of skip_torchrec at later stages.

Test Plan:
Tested on APS models:

  buck2 run mode/opt //aps_models/ads/icvr:icvr_launcher_live -- mode=local_ig_fm_uhm_mini model_name=ig_fm_one_sparse_benchmark features=ig_fm_one_sparse_benchmark model=ig_fm_one_sparse_benchmark training.pipeline_type=pt2

commit: 2fb485d9e

torchrec related paths were not skipped.

Differential Revision: D59779153


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

yanboliang reviewed (approved):
Please land it after fixing the lint failure.

Microve committed:
Revise skip torchrec logic (#130783)

Summary:
X-link: https://github.com/pytorch/torchrec/pull/2240

Pull Request resolved: https://github.com/pytorch/pytorch/pull/130783

The previous logic adds skipped files when the file was imported which happens at very early stage. However, we could set skip_torchrec at later stage (e.g, in APS, we set it during the trainer execution). In that case, the skip logic will still take effect since skipped files have been added.

So in this diff, we revise the logic so that it can adapt to changes of skip_torchrec at later stages.

Since we revise the skip torchrec logic, we also update related torchrec tests.

Test Plan:
Tested on APS models:

  buck2 run mode/opt //aps_models/ads/icvr:icvr_launcher_live -- mode=local_ig_fm_uhm_mini model_name=ig_fm_one_sparse_benchmark features=ig_fm_one_sparse_benchmark model=ig_fm_one_sparse_benchmark training.pipeline_type=pt2

commit: 2fb485d9e

torchrec related paths were not skipped.

**Unit tests**
  buck2 test 'fbcode//mode/opt' fbcode//torchrec/distributed/tests:test_pt2_multiprocess

https://www.internalfb.com/intern/testinfra/testrun/7036874643631762

  buck2 test 'fbcode//mode/opt' fbcode//torchrec/distributed/tests:test_pt2

https://www.internalfb.com/intern/testinfra/testrun/16888498645009972

  buck2 test 'fbcode//mode/opt' fbcode//torchrec/fb/ads/distributed/tests:test_pt2_multiprocess

https://www.internalfb.com/intern/testinfra/testrun/12384899017649985

  buck2 test 'fbcode//mode/opt' fbcode//torchrec/distributed/tests:test_comm

https://www.internalfb.com/intern/testinfra/testrun/2251800056679050

Reviewed By: IvanKobzarev, yanboliang

Differential Revision: D59779153
